### PR TITLE
Update Changelog of 3.5 and 3.4

### PR DESCRIPTION
**Pull request by deanurag**
Updated Versions in the changelog file .
Part of https://github.com/etcd-io/etcd/issues/19000

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: CHANGELOG/CHANGELOG-3.5.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ Previous change logs can be found at [CHANGELOG-3.4](https://github.com/etcd-io/
 
 ### Package `clientv3`
 - Fix [runtime panic that occurs when KeepAlive is called with a Context implemented by an uncomparable type](https://github.com/etcd-io/etcd/pull/18937)
+- Compile binaries using [go 1.22.10](https://github.com/etcd-io/etcd/pull/19004).
```
Comment (ahrtr) [inline CHANGELOG/CHANGELOG-3.5.md:13]: Please add this item under `## Dependencies`

Comment (ahrtr) [reply CHANGELOG/CHANGELOG-3.5.md:15]: There is already a section `## Dependencies` below (line 20), please add this item there.

Comment (deanurag) [reply CHANGELOG/CHANGELOG-3.5.md:15]: below line 20 , is the section of etcd grpc-proxy, do you want me to add the same thing there ?

**Comment by ahrtr** [comment]
@deanurag please try commands something like below,

```
git reset --soft HEAD~1
git checkout -- scripts/release.sh
git add .
git commit --signoff -m "Update changelog for both 3.4 and 3.5 to cover the go version bumping"

# manually sync your github repo with upstream: on your fork repo deanurag/etcd page, click "sync fork"
git checkout main
git pull

git checkout update/ChangeLog
git rebase main
git push origin update/ChangeLog -f

````

**Comment by ahrtr** [comment]
also read https://www.atlassian.com/git/glossary#commands

File: CHANGELOG/CHANGELOG-3.5.md (hunk 1 above)
Comment (ivanvc) [reply CHANGELOG/CHANGELOG-3.5.md:15]: Yes, the section is not nested. The dependencies listed there are not from `etcd grpc-proxy`. Thanks!

File: CHANGELOG/CHANGELOG-3.5.md [hunk 2]
```diff
@@ -11,11 +11,13 @@ Previous change logs can be found at [CHANGELOG-3.4](https://github.com/etcd-io/
 ### Package `clientv3`
 - Fix [runtime panic that occurs when KeepAlive is called with a Context implemented by an uncomparable type](https://github.com/etcd-io/etcd/pull/18937)
 
+
 ### etcd grpc-proxy
 - Add [`tls min/max version to grpc proxy`](https://github.com/etcd-io/etcd/pull/18829) to support setting TLS min and max version.
 
 ## Dependencies
```
Comment (ahrtr) [inline CHANGELOG/CHANGELOG-3.5.md:18]: please feel free to resolve this in a separate PR.
```suggestion
### Dependencies
```

**Review by ahrtr — APPROVED** [review]
Let me squash the commits for you this time.

File: CHANGELOG/CHANGELOG-3.5.md (hunk 2 above)
Comment (ahrtr) [reply CHANGELOG/CHANGELOG-3.5.md:18]: Updated in https://github.com/etcd-io/etcd/pull/19031

**Pull request merged**